test/test_autograd.py

```diff
@@ -1618,20 +1611,22 @@ def test_reentrant(self):
 
         class Reenter(Function):
             @staticmethod
-            def forward(ctx, x_data):
-                ctx.x = Variable(x_data, requires_grad=True)
-                ctx.y = Variable(y_data, requires_grad=True)
-                ctx.output_var = ctx.x * ctx.y
-                return ctx.output_var.data
+            def forward(ctx, x):
+                with torch.enable_grad():
+                    ctx.x = Variable(x.data, requires_grad=True)
+                    ctx.y = Variable(y_data, requires_grad=True)
+                    ctx.output_var = ctx.x * ctx.y
+                return ctx.output_var[:]
 
             @staticmethod
             def backward(ctx, grad_output):
-                ctx.output_var.sum().backward()
+                with torch.enable_grad():
```

Comment:
similar to above, why do you torch.enable_grad here?

Reply:
`backward()` runs with no-grad if `create_graph` is False because you don't care about higher order derivatives. We could run with `create_graph=True` here (the previous behavior), but that saves more history then necessary. It's a little more clean to just use `enable_grad()` so that the `sum()` calls is differentiable, but not he multiplication on the line below.